// Copyright 2019 Shift Cryptosecurity AG
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";
package shiftcrypto.bitbox02;

message CheckSDCardRequest {
}

message CheckSDCardResponse {
  bool inserted = 1;
}

message DeviceInfoRequest {
}

message DeviceInfoResponse {
  message Bluetooth {
    // Hash of the currently active Bluetooth firmware on the device.
    bytes firmware_hash = 1;
    // Firmware version, formated as an unsigned integer "1", "2", etc.
    string firmware_version = 2;
    // True if Bluetooth is enabled
    bool enabled = 3;
  }
  string name = 1;
  bool initialized = 2;
  string version = 3;
  bool mnemonic_passphrase_enabled = 4;
  uint32 monotonic_increments_remaining = 5;
  // From v9.6.0: "ATECC608A" or "ATECC608B".
  string securechip_model = 6;
  // Only present in Bluetooth-enabled devices.
  optional Bluetooth bluetooth = 7;
}

message InsertRemoveSDCardRequest {
  enum SDCardAction {
    REMOVE_CARD = 0;
    INSERT_CARD = 1;
  }
  SDCardAction action = 1;
}

message ResetRequest {}

message SetDeviceLanguageRequest {
    string language = 1;
}

message SetDeviceNameRequest {
    string name = 1;
}

message SetPasswordRequest {
    bytes entropy = 1;
}
